homeassistant/components/advantage_air/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""Advantage Air climate integration."""
+
+from datetime import timedelta
+import logging
+
+from advantage_air import ApiError, advantage_air
+
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import ADVANTAGE_AIR_RETRY, DOMAIN
+
+ADVANTAGE_AIR_SYNC_INTERVAL = 15
+ADVANTAGE_AIR_PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up AdvantageAir."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up AdvantageAir Config."""
+    ip_address = config_entry.data[CONF_IP_ADDRESS]
+    port = config_entry.data[CONF_PORT]
+    api = advantage_air(ip_address, port, ADVANTAGE_AIR_RETRY)
```

Comment:
The Advantage Air hardware's API doesn't support the use of a persistent sessions. I did try using the Home Assistant session initially but it raised frequent errors about the session being closed and it caused reliability issues. I believe this is because the Android App we interface with isn't a fully featured web server.

Reply:
Do you have the log with the errors?

Reply:
https://github.com/Bre77/myair/issues/3
"Cannot write to closing transport"

Reply:
The session should be used with an async context manager `async with`. See eg:

https://github.com/home-assistant-libs/pydroid-ipcam/blob/1c70a1b440c5f918ac305b79c6872182fc121784/pydroid_ipcam/__init__.py#L73

That wasn't done here:
https://github.com/Bre77/myair/blob/497d0d630f9aafda0b2712b2a6c82045ddd6d468/custom_components/myair/__init__.py#L52

Reply:
Appreciate the depth you went to here to identify the true root cause. I'll rework the library to use a passed in session, similar how I did for the Splunk library.

Reply:
This has now been pushed. Should make fixing the tests easier now that I can mock the session.